tests/unit/s2n_connection_test.c

```diff
@@ -126,7 +126,7 @@ int main(int argc, char **argv)
      */
     {
         /* Carefully consider any increases to this number. */
-        const uint16_t max_connection_size = 4150;
+        const uint16_t max_connection_size = 4176;
```

Comment:
I think I'm missing some context, but why does adding an extension array require us to increase this number? I also ask because of the in-line comment above

Reply:
I revised my original PR description to attempt to answer this better.